File: query-service/src/main/java/ru/telemetry/query/model/DeviceView.java
package ru.telemetry.query.model;

import java.time.Instant;

/**
 * Ответ GET /devices — реестр устройств (имя/группа, Postgres read-реплика) объединён с
 * текущим состоянием из Redis. Устройство, ни разу не приславшее данные или "протухшее" по TTL
 * в Redis, всё равно попадает в список (status=OFFLINE, координаты null) — иначе оно просто
 * молча исчезает, и на dashboard не видно, что оно вообще пропало.
 */
public record DeviceView(
        String deviceId,
        String name,
        String groupName,
        boolean active,
        DeviceStatus status,
        Double lat,
        Double lon,
        Double speedKmh,
        Instant recordedAt
) {
}

File: dashboard/src/app/models/device-view.ts
// Контракт ответа query-service (GET /devices) — должен оставаться совместим с
// ru.telemetry.query.model.DeviceView на бэкенде.
export type DeviceStatus = 'ONLINE' | 'STALE' | 'OFFLINE';

export interface DeviceView {
  deviceId: string;
  name: string;
  groupName: string;
  active: boolean;
  status: DeviceStatus;
  lat: number | null;
  lon: number | null;
  speedKmh: number | null;
  recordedAt: string | null;
}
